documentation/src/main/doc/modules/kafka/examples/inbound/KafkaRecordBatchExample.java

```diff
@@ -0,0 +1,44 @@
+package inbound;
+
+import java.util.concurrent.CompletionStage;
+
+import javax.enterprise.context.ApplicationScoped;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.clients.consumer.ConsumerRecords;
+import org.apache.kafka.common.TopicPartition;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+
+import io.smallrye.reactive.messaging.kafka.KafkaRecordBatch;
+import io.smallrye.reactive.messaging.kafka.KafkaRecord;
+import io.smallrye.reactive.messaging.kafka.api.IncomingKafkaRecordMetadata;
+
+@ApplicationScoped
+public class KafkaRecordBatchExample {
+
+    @SuppressWarnings("unchecked")
+    // tag::code[]
+    @Incoming("prices")
+    public CompletionStage<Void> consumeMessage(KafkaRecordBatch<String, Double> records) {
+        for (KafkaRecord<String, Double> record : records) {
+            record.getMetadata(IncomingKafkaRecordMetadata.class).ifPresent(metadata -> {
+                int partition = metadata.getPartition();
+                long offset = metadata.getOffset();
+                ConsumerRecord<String, Double> consumerRecord = metadata.getRecord();
```

Comment:
This is equivalent to existing `record`, right?

Reply:
yes `KafkaRecord` is the `Message` wrapper, `ConsumerRecord` is the Kafka client type.